### PR TITLE
feat(directory): Orphanet hierarchy

### PR DESCRIPTION
**Pull request by dtroelofsprins**
how to test:
https://preview-emx2-pr-3480.dev.molgenis.org
- Model + DiseaseTypes update: Go to the preview, login and check table [DiseaseTypes](https://preview-emx2-pr-3480.dev.molgenis.org/directory-demo/tables/#/DiseaseTypes). Please note that the Diagnosis available filter is broken (as expected) due to these changes.

todo:
- [na] updated docs in case of new feature
- [na] added tests

### CODE REVIEW

**Comment by dtroelofsprins** [comment]
[#10680 Add parent child information to Orphanet codes in the diseases table](https://trello.com/c/XCvmHsMj/1583-10680-add-parent-child-information-to-orphanet-codes-in-the-diseases-table)

**Comment by esthervanenckevort** [comment]
Ik zie bij als parents bij ORPHA:443909 (Hereditary nonpolyposis colon cancer) de volgende codes geeft:

- ORPHA:271835 (Genetic digestive tract tumor)
- ORPHA:425003 (Inherited digestive cancer-predisposing syndrome)
- ORPHA:557492 (group of disorders)
- ORPHA:589746 (Inherited gynecological cancer-predisposing syndrome)
 
En bij ORPHA:144 (Lynch syndrome) de volgende codes als parent:

- ORPHA:377788 (disease)
- ORPHA:443909 (Hereditary nonpolyposis colon cancer)
- ORPHA:557493 (disorder)

Ik denk dat we ORPHA:377788 (disease), ORPHA:557493 (disorder) en ORPHA:557492 (group of disorders) niet nodig hebben als parents. Je wilt degene die als een van de parents ORPHA:557492 (group of disorders) aan het begin van je boomstructuur hebben staan. Anders moet je altijd beginnen met group of disorders open te klikken en heb je onder disorder en disease nog steed heel lange lijsten van dezelfde codes.

**Review by esthervanenckevort — COMMENTED** [review]
Spot checked the hierarchy and it looks good.

Now we still need a developer to fix the diagnosis filter, until then it cannot be merged.

File: data/biobank-directory/ontologies/DiseaseTypes.csv
Comment (esthervanenckevort) [inline data/biobank-directory/ontologies/DiseaseTypes.csv:1]: A couple of entries that still need to be filtered out:
- ORPHA:top
- ORPHA:https://creativecommons.org/licenses/by/4.0/
- ORPHA:C001 clinical entity
- ORPHA:557493 disorder
- ORPHA:557492 group of disorders
- ORPHA:377788 disease
- ORPHA:557494 subtype of a disorder (all entries below that should still be accessible through the main type)

The inactive clinical entity hierarchy should be kept, so people can still search for codes that are no longer active, but might have been used in the past.

File: apps/directory/src/filter-config/filterOptions.ts [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+          new QueryEMX2("graphql")
+            .table(sourceTable)
+            .select(selection)
+            .orderBy(sourceTable, sortColumn, sortDirection)
+            .execute()
+            .then((response: any) => {
+              const itemsSplitByOntology = getItemsSplitByOntology(
+                response[sourceTable],
+                ontologyIdentifiers
+              );
```
Comment (chinook25) [inline apps/directory/src/filter-config/filterOptions.ts:142]: Comment for code reviewer: changes start with these lines

**Comment by chinook25** [comment]
Code changes also seem to resolve #3502

**Review by esthervanenckevort — CHANGES_REQUESTED** [review]
Code changes look fine. Some remaining issues:
- I still see the issue in #3502 on the preview server. Either fix it in this PR or fix it separately and make sure the issue is not closed with this PR.
- The search only shows the top level code(s) under which the matching code is 
- The Orphanet data still contains a few items that should be removed. See the separate comment about that. @dtroelofsprins

**Comment by chinook25** [comment]
>     * I still see the issue in [Diagnosis search hangs when searching for leukemia #3502](https://github.com/molgenis/molgenis-emx2/issues/3502) on the preview server. Either fix it in this PR or fix it separately and make sure the issue is not closed with this PR.
Removed the comment, let's let that in a separate PR in order to not bloat this one too much. 

>     * The search only shows the top level code(s) under which the matching code is
Can you elaborate? Search behaviour is the same as it currently is on master, and works for me. If we want it changed it's better to make that it's own story.

**Comment by esthervanenckevort** [comment]
> Can you elaborate? Search behaviour is the same as it currently is on master, and works for me. If we want it changed it's better to make that it's own story.

When you search for "Lynch" you see the three main categories under which "Lynch" falls, but you don't see the matching code itself, because the tree is in a collapsed state. This makes it unclear for a user if there was a match found and requires additional steps of digging down the tree to select "Lynch". Showing the parents is good, but I think it is better if we open the tree so you can also see the match. I'll create a separate story for that.

**Review by esthervanenckevort — APPROVED** [review]
The Orphanet code hierarchy is now also clean.

**Review by konstantina-gkp — APPROVED** [review]

**Pull request merged**